packages/realm-server/migrations/1728561353454_create-user-billing.js

```diff
@@ -0,0 +1,128 @@
+exports.up = (pgm) => {
+  pgm.createTable('users', {
+    id: 'id',
+    matrix_user_id: { type: 'varchar', notNull: true, unique: true },
+    stripe_customer_id: { type: 'varchar' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('plans', {
+    id: 'id',
+    name: { type: 'varchar', notNull: true },
+    monthly_price: { type: 'numeric', notNull: true },
+    credits_included: { type: 'integer', notNull: true },
+    overage_fee_per_credit_in_usd: { type: 'numeric' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('billing_cycles', {
+    id: 'id',
+    subscription_id: { type: 'integer' },
+    period_start: { type: 'timestamp', notNull: true },
+    period_end: { type: 'timestamp', notNull: true },
+  });
+
+  pgm.createTable('subscriptions', {
+    id: 'id',
+    user_id: { type: 'integer', references: 'users(id)' },
+    plan_id: { type: 'integer', references: 'plans(id)' },
+    started_at: { type: 'timestamp', notNull: true },
+    ended_at: { type: 'timestamp', notNull: true },
+    status: { type: 'varchar', notNull: true },
+    stripe_subscription_id: { type: 'varchar', notNull: true },
+    current_billing_cycle_id: {
+      type: 'integer',
+      references: 'billing_cycles(id)',
+    },
```

Comment:
is there a reason why we create an actual entry for this and not derive it from the billing_cycles table? could this get out of sync otherwise?

Reply:
I added it for convenience so that the queries don't grow too large, otherwise we could also simply get it by querying the most recent cycle from the table